### DISCUSSION TITLE
Will cores with 1k D_MEM be supported?

### DISCUSSION THREAD

**Discussion by DL8** [amichai-bd/riscv-multi-core-lotr#67] [Q&A] [answered]
I changed the implementation of the library and it won't work if there is less than 1k D_MEM in the core. Is it safe to assume that it won't be supported?

**Comment by amichai-bd** [comment]
For the GPC_4T core I think its safe to assume 1K of DMEM.

But incase we design a new core with a single thread - i want to support D_MEM as small as 512Bytes.

**Comment by DL8** [comment, marked as answer]
Just to clarify: ***at least*** 1K of DMEM.

The current layout can be significantly simplified if there is only one thread, so I believe 512B can be supported in that case.

**Reply by amichai-bd** [reply]
Great!
So yes - safe to assume that D_MEM is greater than 1K in the GPC_4T core. (4 threads)
